src/NzbDrone.Core/IndexerSearch/Definitions/MovieSearchCriteria.cs

```diff
@@ -23,5 +23,31 @@ public override bool RssSearch
         }
 
         public string FullImdbId => ParseUtil.GetFullImdbId(ImdbId);
+
+        public override string SearchQuery
+        {
+            get
+            {
+                if (ImdbId.IsNullOrWhiteSpace())
+                {
+                    if (TmdbId.HasValue)
+                    {
+                        return $"TMDbId:[{TmdbId}]";
+                    }
+                    else if (TraktId.HasValue)
+                    {
+                        return $"TraktId:[{TraktId}]";
+                    }
+                    else
+                    {
+                        return SearchQuery;
+                    }
+                }
+                else
+                {
+                    return $"IMDbId:[{ImdbId}]";
```

Comment:
What happens when multiple IDs are passed in? Should rework your logic to allow for that

Reply:
reworked to include all terms and IDs searched + Season/Episode